src/server/string_family_test.cc

```diff
@@ -92,6 +92,77 @@ TEST_F(StringFamilyTest, Expire) {
   ASSERT_THAT(Run({"incr", "i"}), IntArg(1));
 }
 
+TEST_F(StringFamilyTest, Keepttl) {
+  ASSERT_EQ(Run({"set", "key", "val", "EX", "100"}), "OK");
+  ASSERT_EQ(Run({"set", "key", "val"}), "OK");
+  auto resp = Run({"ttl", "key"});
+  auto actual = get<int64_t>(resp.u);
+  ASSERT_EQ(actual, -1);
+
+  resp = Run({"set", "key", "val", "EX", "200"});
+  ASSERT_EQ(Run({"set", "key", "val", "KEEPTTL"}), "OK");
+
+  resp = Run({"ttl", "key"});
+  actual = get<int64_t>(resp.u);
+
+  EXPECT_TRUE(actual > 0 && actual <= 200);
+}
+
+TEST_F(StringFamilyTest, SetOptionsSyntaxError) {
+  using std::chrono::duration_cast;
+  using std::chrono::milliseconds;
+  using std::chrono::seconds;
+  using std::chrono::system_clock;
+
+  auto now_mill = duration_cast<milliseconds>(system_clock::now().time_since_epoch()).count();
```

Comment:
you should not use real clock for the test and I know you copied it from another place in this file but it's also wrong there.

We use `TEST_current_time_ms` - see generic_family_test, for example